lib/src/main/java/oiccli/State.java

```diff
@@ -0,0 +1,194 @@
+package oiccli;
+
+import oiccli.exceptions.ExpiredToken;
+import oiccli.exceptions.UnknownState;
+import sun.swing.plaf.synth.DefaultSynthStyle;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+public class State {
+
+    private String clientId;
+    private Database db;
+    private int lifetime;
+
+    public State(String clientId, Database db, String dbName, int lifetime) {
```

Comment:
dbName is only used when db is null. So either merge them in a single class or have 2 different constructors.